mcga-parcial2/src/store/slices/products/index.js

```diff
@@ -0,0 +1,98 @@
+import { createSlice } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const initialState = {
+  products: [],
+  status: "idle",
+  error: null
+};
+
+export const productsSlice = createSlice({
+  name: "products",
+  initialState,
+  reducers: {
+    getProducts: state => {
+      state.status = "loading";
+    },
+    getProductsSuccess: (state, action) => {
+      state.status = "succeeded";
+      state.products = action.payload;
+    },
+    getProductsError: (state, action) => {
+      state.status = "failed";
+      state.error = action.payload;
+    },
+    addProduct: (state, action) => {
+      state.products.push(action.payload);
+    },
+    deleteProduct: (state, action) => {
+      state.products = state.products.filter(product => product._id !== action.payload);
+    },
+    updateProduct: (state, action) => {
+      state.products = state.products.map(product => {
+        if (product._id === action.payload._id) {
+          return action.payload;
+        }
+        return product;
+      });
+    }
+  }
+});
+
+export const {
+  getProducts,
+  getProductsSuccess,
+  getProductsError,
+  addProduct,
+  deleteProduct,
+  updateProduct
+} = productsSlice.actions;
+
+export default productsSlice.reducer;
+
+// get all products
+export const fetchProducts = () => async dispatch => {
+  try {
+    dispatch(getProducts());
+    const response = await axios.get("http://localhost:5000/products");
```

Comment:
you are not handling the status of this response, you need to check for the status code and handle it accordingly for the corresponding success and failure of that request